src/lib.rs

```diff
@@ -250,6 +250,38 @@ impl<T> Status<T> {
     }
 }
 
+/// Parser configuration.
+#[derive(Clone, Debug, Default)]
+pub struct ParserConfig {
+    allow_spaces_after_header_name: bool,
+}
+
+impl ParserConfig {
+    /// Sets whether spaces should be allowed after header name.
+    pub fn allow_spaces_after_header_name(&mut self, value: bool) -> &mut Self {
+        self.allow_spaces_after_header_name = value;
+        self
+    }
+
+    /// Parses a request with the given config.
+    pub fn parse_request<'headers, 'buf>(
```

Comment:
I suppose in a general sense, it makes sense to be able to pass some config when parsing requests. Just to be sure, in this specific case, there's no value in allowing spaces before colons in *requests*, since that's illegal and servers should reject it, right?

Reply:
Yeah I don't think I would use it in our proxy.

Reply:
I removed the config for parsing requests, we can always introduce that when we actually need it.